generator/transit_generator.cpp

```diff
@@ -370,14 +444,140 @@ bool GraphData::IsSorted() const
   return v.IsSorted();
 }
 
+void GraphData::ClipLines(vector<m2::RegionD> const & borders)
+{
+  // Set with stop ids with stops which are inside |borders|.
+  set<StopId> stopIdInside;
+  for (auto const & s : m_stops)
+  {
+    if (m2::RegionsContain(borders, s.GetPoint()))
+      stopIdInside.insert(s.GetId());
+  }
+
+  set<StopId> hasNeighborInside;
+  for (auto const & e : m_edges)
+  {
+    if (stopIdInside.count(e.GetStop1Id()) != 0 && stopIdInside.count(e.GetStop2Id()) == 0)
+      hasNeighborInside.insert(e.GetStop2Id());
+    if (stopIdInside.count(e.GetStop2Id()) != 0 && stopIdInside.count(e.GetStop1Id()) == 0)
+      hasNeighborInside.insert(e.GetStop1Id());
+  }
+
+  // Filling |lines| with stops inside |borders|.
+  vector<Line> lines;
+  for (auto const & l : m_lines)
+  {
+    Ranges const & stopIds = l.GetStopIds();
```

Comment:
If the size of vector is always one, why not to replace vector by a variable?

Reply:
size of this vector should be `1` for input (initial json) but after clipping it can be `>1` if line crosses mwm border several times
we use the same data structure (`transit::Line`) for initial and clipped line so I think everything is ok here